lib/ansible/plugins/inventory/ovirt.py

```diff
@@ -0,0 +1,294 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: ovirt
+    plugin_type: inventory
+    short_description: oVirt inventory source
+    requirements:
+      - ovirt-engine-sdk-python >= 4.2.4
+    extends_documentation_fragment:
+        - inventory_cache
+    description:
+      - Get inventory hosts from the ovirt service.
+      - Uses 'ovirt.yml', 'ovirt4.yml', 'ovirt.yaml', 'ovirt4.yaml' YAML configuration file.
+    options:
+      plugin:
+        description: the name of this plugin, it should alwys be set to 'ovirt' for this plugin to recognize it as it's own.
+        required: True
+        choices: ['ovirt']
+      ovirt_url:
+        description: url to ovirt-engine api
+        required: True
+      ovirt_username:
+        description: ovirt authentication user
+        required: True
+      ovirt_password:
+        description: ovirt authentication password
+        required : True
+      ovirt_cafile:
+        description: path to ovirt-engine ca file
+        required: False
+      ovirt_preferred_interface:
```

Comment:
instead of requiring specific options to configure specific host variables, inherit from constructed and allow the user to create his own expressions.

 You assume that the devices are configured following a pattern, while it should be that way in many cases, it isn't always. Users might want to match `ansible_host` to a specific ip range, cluster/iface/tags combos, etc. The compose engine allows much more flexibility in this case.

Reply:
I've updated the plugin to inherit from constructed and allow more flexible expressions. Also improved the vm filtering to be more in the user's control.